fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraLdpIT.java

```diff
@@ -2525,6 +2524,38 @@ public void testPathsWithBrackets() {
     }
 
     @Test
+    public void testConcurrentPuts() throws InterruptedException, IOException {
+        final String parent = getRandomUniqueId();
+        executeAndClose(putObjMethod(parent));
+        final String newResource = parent + "/test";
+        final Runnable updateRunnable = () -> { executeAndClose(putObjMethod(newResource)); };
+        final Thread t1 = new Thread(updateRunnable);
+        final Thread t2 = new Thread(updateRunnable);
+        final Thread t3 = new Thread(updateRunnable);
+        final Thread t4 = new Thread(updateRunnable);
+        t1.start();
+        t2.start();
+        t3.start();
+        t4.start();
+        t1.join();
+        t2.join();
+        t3.join();
+        t4.join();
+
+        try (final CloseableDataset dataset = getDataset(getObjMethod(parent))) {
+            final DatasetGraph graphStore = dataset.asDatasetGraph();
+            final Iterator<Quad> children = graphStore.find(ANY, ANY, CONTAINS.asNode(), ANY);
+            assertTrue("One of the PUTs should have resulted in a child.", children.hasNext());
+            children.next();
+            if (children.hasNext()) {
```

Comment:
I really wanted to include the URI of the unexpected child because it's really enlightening (ie, if it has a [1] in it).  I could use an inline if to avoid a NPE when trying to generate the message String, but it feels wierd to make a String that isn't correct and you know is never going to be seen.